File: skills/scottcjn/grazer-skill/src/clawhub.ts
/**
 * ClawHub Integration - Skill Registry with Vector Search
 */

import axios, { AxiosInstance } from 'axios';

export interface ClawHubSkill {
  id: string;
  name: string;
  description: string;
  author: string;
  version: string;
  downloads: number;
  tags: string[];
  platforms: string[];
  npm_package?: string;
  pypi_package?: string;
  github_repo?: string;
}

export class ClawHubClient {
  private http: AxiosInstance;
  private token?: string;

  constructor(token?: string) {
    this.token = token;
    this.http = axios.create({
      baseURL: 'https://clawhub.ai/api/v1',
      timeout: 15000,
      headers: {
        'User-Agent': 'Grazer/1.3.0 (Elyan Labs)',
        ...(token ? { Authorization: `Bearer ${token}` } : {}),
      },
    });
  }

  /**
   * Search skills using vector search
   */
  async searchSkills(query: string, limit = 20): Promise<ClawHubSkill[]> {
    const resp = await this.http.get('/skills/search', {
      params: { q: query, limit },
    });
    return resp.data.skills || [];
  }

  /**
   * Get trending skills
   */
  async getTrendingSkills(limit = 20): Promise<ClawHubSkill[]> {
    const resp = await this.http.get('/skills/trending', {
      params: { limit },
    });
    return resp.data.skills || [];
  }

  /**
   * Get skill by ID
   */
  async getSkill(skillId: string): Promise<ClawHubSkill> {
    const resp = await this.http.get(`/skills/${skillId}`);
    return resp.data;
  }

  /**
   * Publish skill to ClawHub
   */
  async publishSkill(skill: {
    name: string;
    description: string;
    version: string;
    tags: string[];
    platforms: string[];
    npm_package?: string;
    pypi_package?: string;
    github_repo?: string;
  }): Promise<ClawHubSkill> {
    if (!this.token) {
      throw new Error('ClawHub token required for publishing');
    }

    const resp = await this.http.post('/skills', skill);
    return resp.data;
  }

  /**
   * Update skill metadata
   */
  async updateSkill(skillId: string, updates: Partial<ClawHubSkill>): Promise<ClawHubSkill> {
    if (!this.token) {
      throw new Error('ClawHub token required for updates');
    }

    const resp = await this.http.patch(`/skills/${skillId}`, updates);
    return resp.data;
  }

  /**
   * Record download/install
   */
  async recordInstall(skillId: string, platform: 'npm' | 'pypi'): Promise<void> {
    try {
      await this.http.post(`/skills/${skillId}/installs`, { platform });
    } catch (err) {
      // Silent fail
    }
  }
}

export default ClawHubClient;

File: skills/scottcjn/grazer-skill/grazer/clawhub.py
"""
ClawHub Integration - Skill Registry with Vector Search
Python SDK for https://clawhub.ai
"""

import requests
from typing import List, Dict, Optional


class ClawHubClient:
    """Client for the ClawHub skill registry API."""

    BASE_URL = "https://clawhub.ai/api/v1"

    def __init__(self, token: Optional[str] = None, timeout: int = 15):
        self.token = token
        self.timeout = timeout
        self.session = requests.Session()
        self.session.headers.update({"User-Agent": "Grazer/1.3.0 (Elyan Labs)"})
        if token:
            self.session.headers.update({"Authorization": f"Bearer {token}"})

    def search(self, query: str, limit: int = 20) -> List[Dict]:
        """Search skills using vector search."""
        resp = self.session.get(
            f"{self.BASE_URL}/skills",
            params={"search": query, "limit": limit},
            timeout=self.timeout,
        )
        resp.raise_for_status()
        data = resp.json()
        return data.get("items", [])

    def explore(self, limit: int = 20, cursor: Optional[str] = None) -> Dict:
        """Browse latest updated skills."""
        params = {"limit": limit}
        if cursor:
            params["cursor"] = cursor
        resp = self.session.get(
            f"{self.BASE_URL}/skills",
            params=params,
            timeout=self.timeout,
        )
        resp.raise_for_status()
        return resp.json()

    def get_skill(self, slug: str) -> Dict:
        """Get skill details by slug."""
        resp = self.session.get(
            f"{self.BASE_URL}/skills/{slug}",
            timeout=self.timeout,
        )
        resp.raise_for_status()
        return resp.json()

    def trending(self, limit: int = 20) -> List[Dict]:
        """Get trending/popular skills (sorted by downloads)."""
        data = self.explore(limit=limit)
        items = data.get("items", [])
        return sorted(items, key=lambda x: x.get("stats", {}).get("downloads", 0), reverse=True)
